lib/releases.js

```diff
@@ -80,74 +79,84 @@ const contributorsSentence = ({ commits, pullRequests }) => {
 }
 
 const getCategoriesConfig = ({ config }) => {
-  const categoriesConfig = config.categories.reduce((acc, category) => {
-    const labels = category.label ? [category.label] : category.labels
-    labels.forEach(label => (acc[label] = category))
-    return acc
-  }, {})
-
-  const orderedCategories = [
-    UNCATEGORIZED,
-    ..._.flatMap(config.categories, category =>
-      category.label ? category.label : category.labels
-    )
-  ]
-
-  return [categoriesConfig, orderedCategories]
+  const categoriesConfig = [{ title: UNCATEGORIZED }, ...config.categories]
+
+  return categoriesConfig.map(category => {
+    if (!category.labels) category.labels = []
+    if (category.label) {
+      // treat label as labels
+      category.labels.push(category.label)
+      delete category.label
+    }
+    return { ...category, pullRequests: [] }
+  })
 }
 
-const categorizePullRequests = (pullRequests, categories, excludeLabels) => {
-  return pullRequests.reduce(
-    (acc, pullRequest) => {
+const categorizePullRequests = (
+  pullRequests,
+  orderedCategories,
+  excludeLabels
+) => {
+  const categories = orderedCategories.flatMap(category => category.labels)
+
+  // exclude pull requests with exclude labels first than push any uncategorized pull requests to UNCATEGORIZED category
+  const filterPullRequests = pullRequest => {
+    const labels = pullRequest.labels.nodes
+
+    if (labels.some(label => excludeLabels.includes(label.name))) {
+      return false
+    } else if (
+      labels.length === 0 ||
+      !labels.some(label => categories.includes(label.name))
+    ) {
+      orderedCategories[0].pullRequests.push(pullRequest)
```

Comment:
1. We shouldn't mutate inputs as that can cause subtle bugs.
2. This can cause bugs when we decide to change the order of uncategorized PRs. Can you push these to a separate array?

Reply:
created a separate array